src/module_4/notebook.md

```diff
@@ -0,0 +1,2102 @@
+     7   outcome                           float64       
+     8   ordered_before                    float64       
+     9   abandoned_before                  float64       
+     10  active_snoozed                    float64       
+     11  set_as_regular                    float64       
+     12  normalised_price                  float64       
+     13  discount_pct                      float64       
+     14  vendor                            object        
+     15  global_popularity                 float64       
+     16  count_adults                      float64       
+     17  count_children                    float64       
+     18  count_babies                      float64       
+     19  count_pets                        float64       
+     20  people_ex_baby                    float64       
+     21  days_since_purchase_variant_id    float64       
+     22  avg_days_to_buy_variant_id        float64       
+     23  std_days_to_buy_variant_id        float64       
+     24  days_since_purchase_product_type  float64       
+     25  avg_days_to_buy_product_type      float64       
+     26  std_days_to_buy_product_type      float64       
+    dtypes: datetime64[ns](2), float64(19), int64(4), object(2)
+    memory usage: 462.3+ MB
+
+
+    /var/folders/89/kh4sf6hs6xngc9mwq6wfcb_r0000gn/T/ipykernel_55548/2768600364.py:1: SettingWithCopyWarning: 
+    A value is trying to be set on a copy of a slice from a DataFrame.
+    Try using .loc[row_indexer,col_indexer] = value instead
+    
+    See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+      df_selected['order_date'] = pd.to_datetime(df_selected['order_date'])
+    /var/folders/89/kh4sf6hs6xngc9mwq6wfcb_r0000gn/T/ipykernel_55548/2768600364.py:2: SettingWithCopyWarning: 
+    A value is trying to be set on a copy of a slice from a DataFrame.
+    Try using .loc[row_indexer,col_indexer] = value instead
+    
+    See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
+      df_selected['created_at'] = pd.to_datetime(df_selected['created_at'])
+
+
+
+```python
+df.order_id.nunique() > df_selected.order_id.nunique()
+```
+
+
+
+
+    True
+
+
+
+
+```python
+daily_orders = df_selected.groupby('order_date').order_id.nunique()
+daily_orders.head()
+```
+
+
+
+
+    order_date
+    2020-10-05     3
+    2020-10-06     7
+    2020-10-07     6
+    2020-10-08    12
+    2020-10-09     4
+    Name: order_id, dtype: int64
+
+
+
+
+```python
+plt.plot(daily_orders, label="daily orders")
+plt.title("Daily Orders");
+```
+
+
+    
+![png](notebook_files/notebook_9_0.png)
+    
+
+
+To avoid information leakage, like having the same orders in train/test, we make a temporal split. Also, due to the strong temporal component, it makes sense to split data in a chronological way and not randomly.
```

Comment:
This evaluation schema is not incorrect but I would consider: 
* make the 70% 20% 10% split over days instead of over orders (bear in mind that in the train period you are NOT supposed to know how many orders you are gonna have in the future, while with the schema that you are proposing you are implicitly assuming this number, this is not true if you just consider number of days)
* Have a rolling window where you keep moving the train / test to the right and you train and evaluate in different months to see how the model behaves in different periods